app/API/V1/Controllers/UserController.php

```diff
@@ -40,43 +51,80 @@ public function getTTConfig(): array
 
     /**
      * @param SimpleEventContract $event
+     * @throws \Doctrine\DBAL\ConnectionException
      */
-    public function onPostStore (SimpleEventContract $event):void
+    public function onPostStore(SimpleEventContract $event):void
     {
-        if ($event->getEventArgs()['frontEndOptions']['email'] === true) {
-            $result = $event->getEventArgs()['result'];
-            $query = $event->getEventArgs()['params'];
+        $result = $event->getEventArgs()['result'];
 
-            $em = app('em');
+        /** @var UserRepository $userRepo **/
+        $userRepo = $this->getRepo();
+        /** @var User $user **/
+        $user = $userRepo->findOneBy(['id'=>$result['id']]);
 
-            /** @var UserRepository $userRepo */
-            $userRepo = $this->getRepo();
-            $user = $userRepo->findOneBy(['id'=>$result['id']]);
+        /** Set user's default role **/
+        $this->roleRepo->setUserPermissions($user);
 
-            $verification_code = str_random(30);
-            $emailVerification = new EmailVerification();
-            $emailVerification->setVerificationCode($verification_code);
-            $emailVerification->setUser($user->id);
-            $em->persist($emailVerification);
-            $em->flush();
+        /** Create the email verification code **/
+        $emailVerification = $this->emailVerificationRepo->createEmailVerificationCode($user);
 
-            $user->setEmailVerification($emailVerification->getId());
-            $em->persist($user);
-            $em->flush();
-
-            var_dump($user); die;
+        /**
+         * TODO: get this into an email queue
+         */
+        if ($event->getEventArgs()['frontEndOptions']['email'] === true) {
+            Mail::send(
+                'emails.activation',
+                [
+                    'user_name' => $result['name'],
+                    'verification_code' => $emailVerification->getVerificationCode(),
+                    'email' => $result['email'],
+                    'host_name' => env('API_DOMAIN', $_SERVER['HTTP_HOST']),
+                    'code' => base64_encode($result['email'] . '_' . $emailVerification->getVerificationCode())
+                ],
+                function ($m) use ($result) {
+                    $m
+                        ->from('jerome@sweetspotmotion.com', 'SweetSpotMotion')
```

Comment:
Can you put these emails in the env (though they shouldn't be committed as part of this project). These are SSM specific emails, but they should be something that is set in the env file.